[done] sortpom

TheRealMarnes opened the pull request:
Automatically tidy, ordered files > allowing arbitrary file structure

Had to add a maven module, and so I noticed...

stevenschlansker commented:
Hm, this makes an awful lot of changes.  And some of the dependencies were grouped together intentionally.

I'm not a fan of how much more whitespace there is.  Can we do 2 spaces for xml?

TheRealMarnes commented:
> some of the dependencies were grouped together intentionally

I understand, but a lot also get added haphazardly or chronologically, which is of no value... You can't really have half order, it's either ordered or it isn't...

I can remove the sortDependencies property but then we live with having dependencies just thrown together in whichever order. Your and @qualidafial 's choice to make, I guess.

qualidafial commented on .editorconfig:
```diff
@@ -13,3 +13,6 @@ trim_trailing_whitespace = false
 indent_size = 2
+
+[pom.xml]
+indent_size = 2
```
I'm not sure what is the benefit of changing this? Most of our POMs use 4-space tabs.

TheRealMarnes replied:
ask @stevenschlansker ¯\\\_(ツ)\_/¯ #1102 (comment)

qualidafial commented on guava/pom.xml:
```diff
@@ -13,65 +13,65 @@
+    <dependency>
+      <groupId>org.jdbi</groupId>
+      <artifactId>jdbi3-core</artifactId>
+    </dependency>
```
Putting Guava before Jdbi? This feels very naive to me.

The order of dependencies in Maven is a factor in dependency resolution. Reordering them can change the outcome of that resolution.

TheRealMarnes replied:
ask @stevenschlansker ¯\\\_(ツ)\_/¯ https://github.com/jdbi/jdbi/pull/1102#issuecomment-385084836

TheRealMarnes replied:
Hm, did not know that. Well I've disabled sortDependencies anyway as @stevenschlansker asked, so everybody happy?

qualidafial replied:
At the least, dependencies need to go back in the order they were in before.

TheRealMarnes commented:
Ok, so, current config is indent=4 and deps are not sorted

stevenschlansker commented:
we can leave it as 4, that's fine... I thought I saw one going to 8 and reacted negatively :)

qualidafial commented:
@stevenschlansker Any other feedback or concerns before merging?

TheRealMarnes commented:
Ok, so, I took a more attentive look and changed the sortpom config a fair bit:

- removed rules that were set to defaults
- removed the "dangerous" rules that can affect builds

It's pretty minimal now, and the changes to existing poms are also quite minimal. Indentation, some header elements, that's basically it. I think it's good to go and risk-free now.

TheRealMarnes commented:
Maybe we can sort dependencies by scope at least though? It's always nice to have `test` dependencies at the bottom.

TheRealMarnes commented:
done

qualidafial commented:
Thanks!

qualidafial commented:
Hmm, GitHub is saying there are still merge conflicts. Possibly with your other PR that just merged?

TheRealMarnes commented:
merged
